### PR TITLE
Models2

### PR DESCRIPTION
**Pull request by yevhenprudnik**
hello @evgeniidatsiuk, I've added separate collections for friends, comments and likes models. Is it better approach?

### CODE REVIEW

**Comment by evgeniidatsiuk** [comment]
hello @yevhenprudnik I very busy this week, so I can't review. I wiil do it from Monday

Also I want to say that I will wait you on interview ))

See you soon)

**Comment by yevhenprudnik** [comment]
Okay, great)
Can you tell me when exactly the interview will be, or will I be informed
later by HRs?

**Comment by evgeniidatsiuk** [comment]
HR will inform you

File: services/user-service.js [hunk 1]
```diff
@@ -119,139 +120,164 @@ class UserService {
...
       throw ApiError.BadRequest(`You are already friends with ${rejectFriend}`);
     }
-    const incomingRequests = user.incomingRequests;
-    const userIndex = incomingRequests.indexOf(rejectFriend);
-    if (userIndex < 0) {
-      throw ApiError.BadRequest(`User ${rejectFriend} did not send you request`);
-    }
 
-    user.incomingRequests.splice(userIndex, 1);
+    user.incomingRequests.pull(request.id);
```
Comment (evgeniidatsiuk) [inline services/user-service.js:229]: if you have friend model that you don't need to have incomingRequests array in user schema, use friend model instead

File: services/user-service.js [hunk 2]
```diff
@@ -119,139 +120,164 @@ class UserService {
...
     await user.save();
 
-    const acceptFriendIndex = userToReject.outcomingRequests.indexOf(user.username);
-    userToReject.outcomingRequests.splice(acceptFriendIndex, 1);
-    
+    const rejectFriendRequest = userToReject.outcomingRequests.find(el => el.username === user.username);
+    userToReject.outcomingRequests.pull(rejectFriendRequest.id);
```
Comment (evgeniidatsiuk) [inline services/user-service.js:233]: the same for outcomingRequests

**Review by evgeniidatsiuk — CHANGES_REQUESTED** [review]

File: services/post-service.js [hunk 3]
```diff
@@ -22,27 +23,43 @@ class PostService {
...
+      } else {
+        return obj.requesterId;
+      }
+    })
+    const posts = await PostModel.find({ postedBy: {$in: friendsIds}})
+    .populate('postedBy', 'username')
     .skip(postPerPage*page)
     .limit(postPerPage)
-    .sort([['date', -1]]);
+    .sort([['createdAt', -1]]);
```
Comment (evgeniidatsiuk) [inline services/post-service.js:62]: here should be aggregation that will join posts with users and filter by friends

**Review by evgeniidatsiuk — CHANGES_REQUESTED** [review]

File: services/post-service.js (hunk 3 above)
Comment (evgeniidatsiuk) [reply services/post-service.js:62]: https://www.mongodb.com/docs/manual/reference/operator/aggregation/lookup/

**Pull request merged**